tests/MongoDB.Driver.Legacy.Tests/Builders/IndexOptionsBuilderTypedTests.cs

```diff
@@ -173,5 +179,19 @@ public void TestMultipleWeights()
             string expected = "{ \"weights\" : { \"textfield\" : 2, \"idioma\" : 10 } }";
             Assert.Equal(expected, options.ToJson());
         }
+
+        [Fact]
+        public void WildcardProjection_should_return_expected_result()
+        {
+            var options = IndexOptions<TestClass>.SetWildcardProjection(x => x.textfield, true);
+            BsonDocument expected = BsonDocument.Parse("{ \"wildcardProjection\" : { \"textfield\" : 1 } }");
+            options.ToBsonDocument().Should().Be(expected);
+
+            options = IndexOptions<TestClass>
+                .SetWildcardProjection(x => x.textfield, false)
+                .SetWildcardProjection(x => x.details.text, true);
```

Comment:
Do we want an example in the tests that is invalid?

Maybe use true for both "x.textField" and "x.details.text"?

Reply:
I guess it would be nice to show that we can set `false` value too. But you are right that it's not applicable for all fields except `_id`, so I would replace `x.textField` to `x._id`.
Done